jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -926,6 +926,30 @@ impl Pool {
                 .ok_or(ErrorKind::NotFound("Stake pool not found".to_owned()).into()),
         }
     }
+
+    pub fn retirement(&self, context: &Context) -> FieldResult<PoolRetirement> {
+        match &self.data {
+            Some(data) => data
+                .retirement
+                .clone()
+                .map(|retirement| PoolRetirement::from(retirement))
+                .ok_or(
+                    ErrorKind::NotFound("Stake pool has no retirement data associated".to_owned())
```

Comment:
I think it's better to return `None` here (null in graphql). And change the return type to `FieldResult<Option<PoolRetirement>>`.
Currently `Error::NotFound` is being used mostly for search-by-id queries.

Reply:
Makes sense, will do.

Reply:
Makes sense. I updated it. Thanks